samples/CognitiveSearch/Generated/DataSourcesClient.cs

```diff
@@ -45,7 +45,7 @@ internal DataSourcesClient(ClientDiagnostics clientDiagnostics, HttpPipeline pip
         /// <param name="requestOptions"> Parameter group. </param>
         /// <param name="accessCondition"> Parameter group. </param>
         /// <param name="cancellationToken"> The cancellation token to use. </param>
-        public virtual async Task<Response<DataSource>> CreateOrUpdateAsync(string dataSourceName, Enum0 prefer, DataSource dataSource, RequestOptions requestOptions = null, AccessCondition accessCondition = null, CancellationToken cancellationToken = default)
+        public virtual async Task<Response<DataSource>> CreateOrUpdateAsync(string dataSourceName, Enum0 prefer, DataSource dataSource, Models.RequestOptions requestOptions = null, AccessCondition accessCondition = null, CancellationToken cancellationToken = default)
```

Comment:
Where is the configuration that specifies this generated method should take RequestOptions as a parameter?

Reply:
There is no configuration for RequestOptions. I see [CodeModel.yaml](https://github.com/Azure/autorest.csharp/blob/51afb53001beb53b297f8a96c89759169f38f1ac/samples/CognitiveSearch/Generated/CodeModel.yaml#L7135-L7149) file generated [RequestOptions](https://github.com/Azure/autorest.csharp/blob/51afb53001beb53b297f8a96c89759169f38f1ac/samples/CognitiveSearch/Generated/Models/RequestOptions.cs) and put [x-ms-request-id](https://github.com/Azure/azure-rest-api-specs/blob/e606243e5297312781dd7dbfd7ab76d2329cc088/specification/search/data-plane/Azure.Search/preview/2019-05-06-preview/searchservice.json#L75) parameter in it.

@timotheeguerin could you please explain why we create RequestOptions type in CodeModel.yaml?

Reply:
@annelo-msft I talked to @timotheeguerin offline. Thanks @timotheeguerin for looking into this.

@annelo-msft RequestOption type is created because it is mentioned in the swagger [here](https://github.com/Azure/azure-rest-api-specs/blob/e606243e5297312781dd7dbfd7ab76d2329cc088/specification/search/data-plane/Azure.Search/preview/2019-05-06-preview/searchservice.json#L7223).